go/store/nbs/store.go

```diff
@@ -928,27 +901,57 @@ func (nbs *NomsBlockStore) Sources(ctx context.Context) (hash.Hash, []TableFile,
 		return hash.Hash{}, nil, nil
 	}
 
-	fsPersister, ok := nbs.p.(*fsTablePersister)
+	css, err := nbs.chunkSourcesByAddr()
+	if err != nil {
+		return hash.Hash{}, nil, err
+	}
+
 	numSpecs := contents.NumTableSpecs()
 
 	var tableFiles []TableFile
 	for i := 0; i < numSpecs; i++ {
-		info := contents.GetTableSpecInfo(i)
-
-		if ok {
-			tf := NomsBlockStoreTableFile{
-				NomsBlockStoreTableFileInfo: NomsBlockStoreTableFileInfo{info: info},
-				dir:                         fsPersister.dir,
-			}
-			tableFiles = append(tableFiles, tf)
-		} else {
-			tableFiles = append(tableFiles, NomsBlockStoreTableFileInfo{info: info})
+		info := contents.getSpec(i)
+		cs, ok := css[info.name]
+		if !ok {
+			return hash.Hash{}, nil, errors.New("manifest referenced table file for which there is no chunkSource.")
 		}
+		tf := tableFile{
+			info: info,
+			open: func() (io.ReadCloser, error) {
+				r, err := cs.reader(context.TODO())
+				if r != nil {
+					return ioutil.NopCloser(r), err
+				} else {
+					return nil, err
+				}
```

Comment:
more idiomatic to do:

```suggestion
				if err != nil {
					return nil, err
				}
				
				return ioutil.NopCloser(r), err
```

Is there a case where r could be nil and err would also be nil that you are worried about?  If so seems like that should also return an error.